relay-server/src/services/buffer/stacks_manager/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::{Envelope, EnvelopeStack};
+
+pub mod memory;
+pub mod sqlite;
+
+/// Enum representing the current capacity of the [`StacksManager`] to accept new [`Envelope`]s.
+pub enum Capacity {
+    Free,
+    Full,
+}
+
+/// A provider of [`EnvelopeStack`] instances that is responsible for creating them.
+pub trait StacksManager: std::fmt::Debug {
```

Comment:
Renamed to `StacksManager` to make it more meaningful w.r.t. the mansions of the trait implementations.

Reply:
The thing is that for me they are two separate entities, it's a coincidence that some of them forward the methods. 

The dependency is:
```mermaid
flowchart TD
	StacksManager --> EnvelopeStack --> EnvelopeStore
```

Reply:
`StacksManager` holds a reference to `EnvelopeStore`, right? So there is already a coupling between the manager and the store:

```mermaid
classDiagram
StacksManager "1" o-- "1" EnvelopeStore
StacksManager --> "creates" EnvelopeStack
EnvelopeStack "n" o-- "1" EnvelopeStore
```

IMO this can be simplified to

```mermaid
classDiagram
EnvelopeStack o-- "uses" EnvelopeStore
EnvelopeStore --> "creates" EnvelopeStack
```

(store creates stack, stack uses store)

Reply:
I feel like having the store interact with the stack is not ideal, for me the store is just offering data to some consumers, the stack is dealing with the data in a domain-specific way and the stack manager deals with creating stacks.